version: '3.8'

networks:
  internal_net:
    external: false

services:
  fastapi-app:
    build:
      context: ./backend
      dockerfile: Dockerfile.controller
    ports:
      - "8000:8000"
    env_file:
      - .env
    depends_on:
      rabbitmq:
        condition: service_healthy
      mongodb:
        condition: service_started
    networks:
      - internal_net

  rabbit-consumer:
    build:
      context: ./backend
      dockerfile: Dockerfile.rabbit
    env_file:
      - .env
    depends_on:
      rabbitmq:
        condition: service_healthy
      mongodb:
        condition: service_started
    networks:
      - internal_net


  mongodb:
    container_name: mongo
    image: mongo:latest
    ports:
      - "27017:27017"
    volumes:
      - mongodb_data:/data/db
    networks:
      - internal_net

  rabbitmq:
    image: rabbitmq:3.10.7-management
    container_name: rabbitmq
    hostname: rabbitmq
    restart: unless-stopped
    healthcheck:
      test: rabbitmq-diagnostics -q ping
      interval: 30s
      timeout: 20s
      retries: 5
    env_file:
      - .env
    ports:
      - "15672:15672"  # admin-panel
      - "5672:5672"    # amqp
    volumes:
      - ./rabbitmq/data:/var/lib/rabbitmq
    networks:
      - internal_net

volumes:
  mongodb_data:
